src/main/java/htsjdk/samtools/util/SortingCollection.java

```diff
@@ -338,8 +338,48 @@ public void cleanup() {
                                                        final Comparator<T> comparator,
                                                        final int maxRecordsInRAM) {
 
-        final File tmpDir = new File(System.getProperty("java.io.tmpdir"));
+        final Path tmpDir = IOUtil.getDefaultTmpDirPath();
+        return new SortingCollection<T>(componentType, codec, comparator, maxRecordsInRAM, tmpDir);
+    }
+
+    /**
+     * Syntactic sugar around the ctor, to save some typing of type parameters
+     *
+     * @param componentType Class of the record to be sorted.  Necessary because of Java generic lameness.
+     * @param codec For writing records to file and reading them back into RAM
+     * @param comparator Defines output sort order
+     * @param maxRecordsInRAM how many records to accumulate in memory before spilling to disk
+     * @param tmpDir Where to write files of records that will not fit in RAM
+     */
+    public static <T> SortingCollection<T> newInstance(final Class<T> componentType,
+            final SortingCollection.Codec<T> codec,
+            final Comparator<T> comparator,
+            final int maxRecordsInRAM,
+            final Path... tmpDir) {
         return new SortingCollection<T>(componentType, codec, comparator, maxRecordsInRAM, tmpDir);
+
+    }
+
+    /**
+     * Syntactic sugar around the ctor, to save some typing of type parameters
+     *
+     * @param componentType Class of the record to be sorted.  Necessary because of Java generic lameness.
+     * @param codec For writing records to file and reading them back into RAM
+     * @param comparator Defines output sort order
+     * @param maxRecordsInRAM how many records to accumulate in memory before spilling to disk
+     * @param tmpDirs Where to write files of records that will not fit in RAM
+     */
+    public static <T> SortingCollection<T> newInstanceFromPaths(final Class<T> componentType,
+            final SortingCollection.Codec<T> codec,
+            final Comparator<T> comparator,
+            final int maxRecordsInRAM,
+            final Collection<Path> tmpDirs) {
+        return new SortingCollection<T>(componentType,
```

Comment:
untested

Reply:
Again, this is just a constructor delegating in other...